kafka-client/src/main/java/io/jenkins/plugins/remotingkafka/KafkaProducerClient.java

```diff
@@ -0,0 +1,31 @@
+package io.jenkins.plugins.remotingkafka;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+
+import java.util.Properties;
+import java.util.logging.Logger;
+
+public class KafkaProducerClient {
+    private static final Logger LOGGER = Logger.getLogger(KafkaProducerClient.class.getName());
+    private static KafkaProducerClient instance;
+    private Producer<String, byte[]> byteProducer;
+
+    private KafkaProducerClient() {
+
+    }
+
+    public static KafkaProducerClient getInstance() {
```

Comment:
Ideally we need different logic for Plugin and for Agent here (`@Extension` for plugin and static for agent)

Reply:
But IIUC this will create code duplication if we separate because the way we use producer/consumer is almost the same in both agent and plugin